fedbiomed/common/training_args.py

```diff
@@ -334,6 +353,9 @@ def default_scheme(cls) -> Dict:
             },
             "share_persistent_buffers": {
                 "rules": [bool], "required": False, "default": True
+            },
+            "random_seed": {
+                "rules": [cls.optional_type(typespec=int, argname='random_seed')], "required": True, "default": None
```

Comment:
Hi @sharkovsky, 

I didn't understand why you did need create `optional_type` method. Isn't `required=False` for optional arguments?

Reply:
Sadly no 😢 

There are two confusing things
## 1. `required` does not mean the user is required to provide the value

The way it is implemented, `required=True|False` means that it is required during the execution of the round, not that the user is required to provide it. 
There is an interaction with the `default` field as well, and the fact that inside the round we initialize the TrainingArgs using `TrainingArgs(self.training_kwargs, only_required=False)`. 

I will give some examples to illustrate:

| `required` | `default` exists? | user provided the value? | behaviour as implemented | 
| ---  | --- | --- | --- |
| True | yes | no | default value is substituted |
| True | no | no | error |
| False | yes | no | default value is substituted |
| False | no | no | nothing is done |

So basically if I put `required=True`, then even if the user doesn't specify a value the default one will be provided (so `required` does not mean it is required from the user)

## 2. `rules` are independent of `required`
Even if `required` had the expected meaning, the `rules` would still be checked: if I put `"rules": int, ..., "default": None` it will fail because `None` is not an `int`.